### PR TITLE
Adding Forgot Password Flow

### PR DESCRIPTION
**Pull request by ZoroarkZark**
This pull request is to add the screens and UI of the forgot password and reset password pages. At the moment, I will soon be using the verification code screen for the flow to be complete, and the rest of the functionality will need to be done in tandem with the backend.

### CODE REVIEW

File: auth/forgot-password/Environment.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import {
+  cognitoFormatEmailOrPhoneNumber,
+  isCognitoErrorWithCode
+} from "@auth/CognitoHelpers"
+import { Auth } from "@aws-amplify/auth"
+import { EmailAddress, Password, USPhoneNumber } from ".."
+
+export type ForgotPasswordResult = "success" | "invalid-email"
```
Comment (ZoroarkZark) [inline auth/forgot-password/Environment.ts:8]: invalid-phone-number

File: auth/forgot-password/ForgotPasswordForm.test.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import { useForgotPasswordForm } from "./ForgotPasswordForm"
+describe("Forgot Password Form tests", () => {
+  describe("Verify Email logic tests", () => {
+    const queryClient = createTestQueryClient()
+    beforeEach(() => queryClient.clear())
+
+    afterAll(() => {
+      queryClient.resetQueries()
+      queryClient.clear()
+    })
```
Comment (ZoroarkZark) [inline auth/forgot-password/ForgotPasswordForm.test.tsx:16]: Can delete, due to the TestQueryClientProvider component providing one for you

File: auth/forgot-password/ForgotPasswordForm.tsx [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                : "invalid-phone-number"
+          } as const
+        }
+      },
+      {
+        onSuccess: (_, args) => onSuccess(args.emailOrPhoneNumber),
+        onError: () => {
+          Alert.alert(
+            "Whoops",
+            "Sorry, something went wrong when trying to change your password. Please try again.",
```
Comment (ZoroarkZark) [inline auth/forgot-password/ForgotPasswordForm.tsx:65]: Be more generic, don't need to be too specific

File: auth/forgot-password/ForgotPasswordNavigation.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+      <stack.Screen name="resetPassword">
+        {(props: any) => (
+          <ResetPasswordScreen
+            {...props}
+            submitResettedPassword={env.submitResettedPassword}
+          />
+        )}
+      </stack.Screen>
+    </>
```
Comment (ZoroarkZark) [inline auth/forgot-password/ForgotPasswordNavigation.tsx:79]: Hide the titles + add options
ForgotPass + Verify: X to close
Reset: Chevron to go back

File: auth/forgot-password/ResetPasswordForm.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  } else if (!validatedNewPassword) {
+    return "weak-new-password" as const
+  } else {
+    return undefined
+  }
+}
+
+const newPasswordErrorMessage = (error?: "weak-new-password" | "empty") => {
+  if (error === "weak-new-password") {
+    return "Your password should be at least 8 characters, and contain at least 1 capital letter, number, and special character."
```
Comment (ZoroarkZark) [inline auth/forgot-password/ResetPasswordForm.tsx:78]: Your password should be at least 8 characters ONLY

**Review by mhayes853 — APPROVED** [review]

**Pull request merged**